src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1386,4 +1405,31 @@ public static void onAdvancement(EntityPlayerMP player, Advancement advancement)
     {
         MinecraftForge.EVENT_BUS.post(new AdvancementEvent(player, advancement));
     }
+    
+    public static void loadFunctions(Map<ResourceLocation, FunctionObject> functions)
+    {
+        Map<ResourceLocation, FunctionObject> extraFunctions = new HashMap<>();
+        MinecraftForge.EVENT_BUS.post(new FunctionLoadEvent(extraFunctions));
+        functions.putAll(extraFunctions);
+    }
+    
+    public static boolean loadStructure(ResourceLocation id, Map<String, Template> templates, boolean flag)
+    {
+        String path = id.getResourcePath();
+        Template structure = templates.get(path);
+        StructureLoadEvent event = new StructureLoadEvent(id, structure);
+
+        if (MinecraftForge.EVENT_BUS.post(event) || event.getStructure() == null)
```

Comment:
Why support removing the template? And again, another thing that is moving data to code. Stop that.

Reply:
I could change this so the event isn't cancellable, and if null is returned it won't try to remove the structure. So if it tries to load a structure that doesn't exist, null is passed into the event, and at the end if it is still null it won't do anything.